ray-project/ray
# [Jobs][Logs] Write JobSupervisor logs to file

## Why are these changes needed?
- Write `JobSupervisor` system logs to file at `jobs/supervisor-{job_id}.log` and stream to stderr so it can continue to be picked up in the `worker-*.err` logs
- This allows for better organization and persistence of the `JobSupervisor` logs

## Manual Testing
- Start ray cluster with `ray start --head` and submit multiple jobs with `RAY_ADDRESS='http://127.0.0.1:8265' ray job submit --working-dir . -- python script.py`
- Output of the jobs should be visible in the console and in `/tmp/ray/session_latest/logs/worker-*.err`
- Files are also created in `/tmp/ray/session_latest/jobs` with only the system logs
```
% cat session_latest/logs/jobs/supervisor-raysubmit_r2ymAYrkF8z2JrVt.log 
Submitting job with RAY_ADDRESS = 127.0.0.1:6379
Job raysubmit_r2ymAYrkF8z2JrVt entrypoint command exited with code 0
```

## Related issue number



## Checks

           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- Testing Strategy
